moldesign/models/lammps_model.py

```diff
@@ -0,0 +1,280 @@
+# Copyright 2016 Autodesk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import
+
+import moldesign as mdt
+import moldesign.units as u
+
+from moldesign import compute
+from moldesign import forcefields as ff
+
+from .base import EnergyModelBase
+
+try:
+    from lammps import lammps, PyLammps
+    # WARNING: this is the real library, not our interface - this works because of absolute
+    # imports. We should probably rename this interface
+except ImportError:
+    print 'LAMMPS could not be imported; using remote docker container'
+    force_remote = True
+else:  # this should be configurable
+    force_remote = False  # debugging
+
+debugging = True # debugging
+
+import parmed as med
+import tempfile
+import os
+import sys
+from .amber2lammps import *
+
+def exports(o):
+    __all__.append(o.__name__)
+    return o
+__all__ = []
+
+
+@exports
+class LAMMPSPotential(EnergyModelBase):
+    """Creates a LAMMPS system to drive energy calculations.
+    """
+    # NEWFEATURE: need to set/get platform (and properties, e.g. number of threads)
+    DEFAULT_PROPERTIES = ['potential_energy', 'forces']
+
+    def __init__(self, **kwargs):
+        super(LAMMPSPotential, self).__init__(**kwargs)
+        self._prepped = False # is the model prepped?
+        self._lammps_data_path = None
+
+        # For temporary directory
+        # create temporary file system
+        # Create temporary filesystem
+        self.tmpdir = tempfile.mkdtemp()
+        self.saved_umask = os.umask(0077)
+        self.files = []
+
+        self.lammps_system = None # Basic LAMMPS system for this molecule
+        self.unit_system = None
+
+    def __exit__(self, exc_type, exc_value, traceback):
```

Comment:
Have you checked to see if this method gets executed? I think you'll want to rename it to `__del__` if you want it to run when this object is garbage-collected.